addons/microsoft_calendar/models/calendar.py

```diff
@@ -74,13 +74,32 @@ def create(self, vals_list):
         if self._check_microsoft_sync_status() and not notify_context and recurrency_in_batch:
             self._forbid_recurrence_creation()
 
-        # for a recurrent event, we do not create events separately but we directly
-        # create the recurrency from the corresponding calendar.recurrence.
-        # That's why, events from a recurrency have their `need_sync_m` attribute set to False.
-        return super(Meeting, self.with_context(dont_notify=notify_context)).create([
-            dict(vals, need_sync_m=False) if vals.get('recurrence_id') or vals.get('recurrency') else vals
-            for vals in vals_list
-        ])
+        # Create events separately since the sender user may differ between events.
+        events = self.env['calendar.event']
+        users_by_id = self._get_users_from_values_list(vals_list)
+        for vals in vals_list:
+            # When creating event with different organizer, check its sync status and verify if the user is listed as attendee.
+            sender_user, partner_ids = self._get_organizer_user_change_info(vals, users_by_id)
+            if sender_user and sender_user.id and sender_user.id != self.env.user.id:
+                if not self.with_user(sender_user)._check_microsoft_sync_status():
+                    raise ValidationError(_("For creating an event with a different organizer, it is necessary that "
+                                            "the organizer have its Odoo Calendar synced with Outlook Calendar."))
+                elif partner_ids and len(partner_ids) > 0 and sender_user.partner_id.id not in partner_ids:
+                    raise ValidationError(_("It is necessary adding the meeting organizer as attendee during its creation."))
+
+            # for a recurrent event, we do not create events separately but we directly
+            # create the recurrency from the corresponding calendar.recurrence.
+            # That's why, events from a recurrency have their `need_sync_m` attribute set to False.
+            events |= super(Meeting, self.with_user(sender_user).with_context(dont_notify=notify_context)).create([
+                dict(vals, need_sync_m=False) if vals.get('recurrence_id') or vals.get('recurrency') else vals
+            ])
```

Comment:
Ah, indeed. Is it expected that you have multiple different users in this method? One improvement could be to group the vals_list by user_id so that you can take advantage of calling create in batch.